### ISSUE TITLE
Projections produce extra folders and don't cleanup them

### ISSUE THREAD

**Issue by simpl1g** [ClickHouse/ClickHouse#34839]
I'm testing some complex projections and noticed some strange behaviour when creating and dropping them, version 21.11.11

I did

```sql
alter table analytics_grouped_stats_monthly_shard_v4 on cluster ch_shards add PROJECTION day_n
(
  SELECT
    date,
    install_date,
    day_number,
    attribution_partner,
    attribution_network,
    attribution_campaign,
    country_id,
    uniqIf(attribution_id_hashed, is_attributed AND (NOT is_organic_partner_split)) AS device_count_attributed,
    uniqIf(device_hashed, (NOT is_attributed) AND (device_hashed != 18251889321102256092) AND (NOT is_organic_partner_split)) AS device_count_not_attributed,
    if(day_number = 0, sum(install_count), device_count_attributed + device_count_not_attributed) AS cohort,
    sumMapIf([install_date], [install_count], (day_number = 0) AND (install_count > 0)) AS installs_count,
    COLUMNS('_count|_revenue|_spend|_costs|_profit|_share') APPLY sum
  GROUP BY
    date,
    install_date,
    day_number,
    attribution_partner,
    attribution_network,
    attribution_campaign,
    country_id
)
materialize projection
drop projection
add projection day_n_v2
materialize projection
drop projection
add projection day_n_v3
```

In the end I have next folders in folder with part `202110_0_7713_6_7750`
```
day_n_596.proj
...
day_n_860.proj
day_n_v2_706.proj
...
day_n_v2_860.proj
day_n_v3.proj -- correct one
day_n_v3_596.proj
....
day_n_v3_726.proj
```

In folder with part `202111_0_2218_6_6159` there are next projection folders
```
day_n_v2_318.proj
...
day_n_v2_388.proj
day_n_v3.proj -- correct one
day_n_v3_269.proj
....
day_n_v3_327.proj
```

Related errors from log
```
<Error> db.analytics_grouped_stats_monthly_shard_v4: Cannot quickly remove directory /var/lib/clickhouse/data/db/analytics_grouped_stats_monthly_shard_v4/delete_tmp_202110_0_7713_6_7749 by removing files; fallback to recursive removal. Reason: Code: 566. DB::ErrnoException: Cannot rmdir /var/lib/clickhouse/data/db/analytics_grouped_stats_monthly_shard_v4/delete_tmp_202110_0_7713_6_7749, errno: 39, strerror: Directory not empty. (CANNOT_RMDIR) (version 21.11.11.1 (official build))
<Error> db.analytics_grouped_stats_monthly_shard_v4: ~DataPart() should remove part data/db/analytics_grouped_stats_monthly_shard_v4/tmp_mut_202110_0_7713_6_7750/day_n_v3_726.proj/ but its name doesn't start with "tmp" or end with ".tmp_proj". Too suspicious, keeping the part.
```
So I have 2 questions:
 - Is it a bug having folders for deleted projections?
 - Is it a bug or it is ok to have extra folders for current project?

**Comment by amosbird** [comment]
It's a known fixed issue https://github.com/ClickHouse/ClickHouse/issues/33275 . Please update to latest release where projections are enabled by default.

**Comment by simpl1g** [comment]
Thanks @amosbird, any plans to backport this fix? It is only available in 22.2 which is not yet battle tested for production usage

**Comment by den-crane** [comment]
@simpl1g 21.11 is end of life. Only 3 last versions are supported: 22.2, 22.1, 21.12 + LTS (21.8, 21.3)
